File: src/job_market/interfaces/schemas/jobs.py
"""Job-related API schemas."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field

from job_market.domain.jobs.models import WorkMode


class JobIngestRequest(BaseModel):
    source_name: str = Field(..., examples=["manual_upload"])
    source_url: str | None = None
    payload: str = Field(..., min_length=1)


class ManualJobUploadRequest(BaseModel):
    title: str = Field(..., min_length=1)
    description: str = Field(..., min_length=1)
    source_name: str = Field(default="manual_upload")
    source_url: str | None = None
    company_name: str | None = None
    location_text: str | None = None
    salary_text: str | None = None
    work_mode: WorkMode | None = None


class JobResponse(BaseModel):
    job_id: int
    title: str
    source_name: str
    work_mode: str
    location_text: str | None = None
    salary_text: str | None = None
    company_name: str | None = None
    source_url: str | None = None
    posted_at: datetime | None = None
    employment_type: str


class JobLocationResponse(BaseModel):
    country: str | None = None
    region: str | None = None
    city: str | None = None
    raw_text: str | None = None


class JobEnrichmentResponse(BaseModel):
    job_id: int | None = None
    role_family: str | None = None
    seniority: str | None = None
    confidence: float | None = None
    skills: list[str]
    technologies: list[str]
    skill_details: list[dict[str, str | float | bool | None]]
    technology_details: list[dict[str, str | float | None]]
    summary: str | None = None


class JobWithEnrichmentResponse(JobResponse):
    description: str
    location: JobLocationResponse
    seniority_hint: str
    observed_at: datetime | None = None
    enrichment: JobEnrichmentResponse | None = None

File: frontend/src/api/jobs.ts
import { requestJson } from './client'

export type WorkMode = 'remote' | 'hybrid' | 'onsite' | 'unknown'

export type JobSummary = {
  job_id: number
  title: string
  source_name: string
  work_mode: string
  location_text: string | null
  salary_text: string | null
  company_name: string | null
  source_url: string | null
  posted_at: string | null
  employment_type: string
}

export type JobLocation = {
  country: string | null
  region: string | null
  city: string | null
  raw_text: string | null
}

export type JobEnrichment = {
  job_id: number | null
  role_family: string | null
  seniority: string | null
  confidence: number | null
  skills: string[]
  technologies: string[]
  summary: string | null
}

export type JobDetail = JobSummary & {
  description: string
  location: JobLocation
  seniority_hint: string
  observed_at: string | null
  enrichment: JobEnrichment | null
}

export type JobListQuery = {
  limit?: number
  offset?: number
  workMode?: WorkMode
}

export function listJobs(query: JobListQuery, signal?: AbortSignal) {
  const search = new URLSearchParams()
  if (query.limit !== undefined) {
    search.set('limit', String(query.limit))
  }
  if (query.offset !== undefined) {
    search.set('offset', String(query.offset))
  }
  if (query.workMode !== undefined) {
    search.set('work_mode', query.workMode)
  }

  const queryString = search.size > 0 ? `?${search}` : ''
  return requestJson<JobSummary[]>(`/v1/jobs${queryString}`, { signal })
}

export function getJob(jobId: number, signal?: AbortSignal) {
  return requestJson<JobDetail>(`/v1/jobs/${jobId}`, { signal })
}
